a10_octavia/common/a10_types.py

```diff
@@ -0,0 +1,129 @@
+import six
+
+from oslo_config import cfg
+from oslo_config.types import List
+from oslo_log import log as logging
+
+from a10_octavia.common import data_models
+
+LOG = logging.getLogger(__name__)
+
+class ListOfDictOpt(cfg.Opt):
+    """List of Dictionary Options
+
+    Option with ``type`` :class:`ListOfObjects`
+
+    :param name: the option's name
+    :param item_type: type of item (None)
+    :param bounds: if True the value should be inside "[" and "]" pair
+    :param \\*\\*kwargs: arbitrary keyword arguments passed to :class:`Opt`
+
+    """
+    def __init__(self, name, item_type=None, bounds=None, **kwargs):
+        super(ListOfDictOpt, self).__init__(name, type=ListOfObjects(item_type, bounds), **kwargs)
+
+
+class ListOfObjects(List):
+    """ List of Object Type
+
+    The value represents a list of objects of type
+    dictionary. eg - [{}, {}]
+    The value will be validated for their key values for vthunder params.
+
+    :param bounds: if True, value should be inside "[" and "]" pair
+    :param type_name: Type name to be used in the sample config file.
+
+    """
+    def __init__(self, bounds=False, type_name='list of dict values'):
+        super(ListOfObjects, self).__init__(bounds=bounds, type_name=type_name)
+
+    def __call__(self, value):
+        if isinstance(value, (list, tuple)):
+            return list(six.moves.map(self.item_type, value))
+
+        value = value.strip().rstrip(',')
+        if self.bounds:
+            if not value.startswith('['):
+                raise ValueError('Value should start with "["')
+            if not value.endswith(']'):
+                raise ValueError('Value should end with "]"')
+            value = value[1:-1]
+        # Warning - param device_name should not contain spaces
+        value = value.replace(" ", "").replace("\n", "")
+        if value:
+            value = value.split('},{')
+        final_list = []
+        for item in value:
+            item = '{' + item.lstrip('{').rstrip('}') + '}'
+            item = item.replace("\'", "\"")
+            single_dict = json.loads(item)
+            final_list.append(single_dict)
+        return convert_to_rack_vthunder_conf(final_list)
+
+
+def convert_to_rack_vthunder_conf(rack_list):
+    """ Validates for all vthunder nouns for rack devices
+        configurations.
+    """
+    rack_dict = {}
+    validation_flag = False
+    try:
+        for rack_device in rack_list:
+            validation_flag = validate_params(rack_device)
+            if validation_flag:
+                rack_device['undercloud'] = True
+                vthunder_conf = data_models.VThunder(**rack_device)
+                rack_dict[rack_device['project_id']] = vthunder_conf
+            else:
+                LOG.warning('Invalid definition of rack device for '
+                            'project ' + rack_device['project_id'])
+
+    except KeyError as err:
+        LOG.error("Invalid definition of rack device in configuration file."
```

Comment:
If they misconfigure the rack device then we should not boot anything.

Reply:
Had discussion with Omkar on this. If they misconfigure rack conf, it should boot up a vthunder.